client/my-sites/people/delete-user/with-delete-user.js

```diff
@@ -0,0 +1,114 @@
+ */
+import useDeleteUserMutation from 'calypso/data/users/use-delete-user-mutation';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+
+const useSuccessNotice = ( isSuccess, user, isMultisite, siteSlug ) => {
+	const showNotice = React.useRef();
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	React.useEffect( () => {
+		showNotice.current = () => {
+			dispatch(
+				successNotice(
+					isMultisite
+						? translate( 'Successfully removed @%(user)s', {
+								args: { user: user.login },
+								context: 'Success message after a user has been modified.',
+						  } )
+						: translate( 'Successfully deleted @%(user)s', {
+								args: { user: user.login },
+								context: 'Success message after a user has been modified.',
+						  } ),
+					{ id: 'delete-user-notice', displayOnNextPage: true }
+				)
+			);
+			page.redirect( `/people/team${ siteSlug ? `/${ siteSlug }` : '' }` );
+		};
+	} );
+
+	React.useEffect( () => {
+		isSuccess && showNotice.current();
+	}, [ isSuccess ] );
+};
+
+const useErrorNotice = ( isError, user, isMultisite, error ) => {
+	const showDomainErrorNotice = React.useRef();
+	const showGeneralErrorNotice = React.useRef();
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	React.useEffect( () => {
+		showDomainErrorNotice.current = ( domainError ) => {
+			const numDomains = domainError.message.split( ',' ).length;
+			dispatch(
+				errorNotice(
+					translate(
+						'@%(user)s owns following domain used on this site: {{strong}}%(domains)s{{/strong}}. This domain will have to be transferred to a different site, transferred to a different registrar, or canceled before removing or deleting @%(user)s.',
+						'@%(user)s owns following domains used on this site: {{strong}}%(domains)s{{/strong}}. These domains will have to be transferred to a different site, transferred to a different registrar, or canceled before removing or deleting @%(user)s.',
+						{
+							count: numDomains,
+							args: {
+								domains: domainError.message,
+								user: user.login,
+							},
+							components: {
+								strong: <strong />,
+							},
+						}
+					),
+					{ id: 'delete-user-notice' }
+				)
+			);
+		};
+
+		showGeneralErrorNotice.current = () => {
+			dispatch(
+				errorNotice(
+					isMultisite
+						? translate( 'There was an error removing @%(user)s', {
+								args: { user: user.login },
+								context: 'Error message after A site has failed to perform actions on a user.',
+						  } )
+						: translate( 'There was an error deleting @%(user)s', {
+								args: { user: user.login },
+								context: 'Error message after A site has failed to perform actions on a user.',
+						  } ),
+					{ id: 'delete-user-notice' }
+				)
+			);
+		};
+	}, [ dispatch, isMultisite, translate, user.login ] );
+
+	React.useEffect( () => {
+		if ( isError ) {
+			if ( 'user_owns_domain_subscription' === error.error ) {
```

Comment:
The `PeopleNotices` component has one more special case for `error === 'user_has_active_subscriptions'` when deleting a user. It should be migrated into this `useErrorNotice` hook, too.

Reply:
fixed in https://github.com/Automattic/wp-calypso/pull/51841/commits/82bdfa5bf2d93c4ffcd5727a58448651eb029cbb

Reply:
@klimeryk if you could check if that error case still works as expected would be awesome (seems like you added it in https://github.com/Automattic/wp-calypso/pull/51424) 🙏🏻